mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java

```diff
@@ -358,6 +360,29 @@
         }
     }
 
+   @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}" + "/dependencyGraph")
+    public ResponseEntity<Resource<MicoServiceDependencyGraphDTO>> getDependencyGraph(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+                                                                               @PathVariable(PATH_VARIABLE_VERSION) String version) {
+       MicoService micoServiceRoot = getServiceFromDatabase(shortName, version);
+       List<MicoService> micoServices = serviceRepository.getAllDependeesOfMicoService(micoServiceRoot.getShortName(), micoServiceRoot.getVersion());
+       MicoServiceDependencyGraphDTO micoServiceDependencyGraph = new MicoServiceDependencyGraphDTO().setMicoServices(micoServices);
+       LinkedList<MicoServiceDependencyGraphEdgeDTO> micoServiceDependencyGraphEdgeList = new LinkedList<>();
+       for (MicoService micoService : micoServices) {
+           //Request each mico service again from the db, because the dependencies are not included
```

Comment:
What about improving the query to include all required information?

Reply:
Time consuming (for a feature of minor importance) and maybe currently impossible because of constraints from the ogm.
The current solution works and performance is a non issue for us anyway.